axelor-sale/src/main/java/com/axelor/apps/sale/web/SaleOrderLineController.java

```diff
@@ -203,14 +216,61 @@ public void getDiscount(ActionRequest request, ActionResponse response) {
       response.setValue("discountAmount", discounts.get("discountAmount"));
       response.setValue("discountTypeSelect", discounts.get("discountTypeSelect"));
       if (discounts.get("price") != null) {
-        response.setValue("price", discounts.get("price"));
+        if (saleOrderLine.getProduct().getInAti()) {
+          response.setValue("inTaxPrice", discounts.get("price"));
+          response.setValue(
+              "price",
+              saleOrderLineService.convertUnitPrice(
+                  true, saleOrderLine.getTaxLine(), (BigDecimal) discounts.get("price")));
+        } else {
+          response.setValue("price", discounts.get("price"));
+          response.setValue(
+              "inTaxPrice",
+              saleOrderLineService.convertUnitPrice(
+                  false, saleOrderLine.getTaxLine(), (BigDecimal) discounts.get("price")));
+        }
       }
 
     } catch (Exception e) {
       response.setFlash(e.getMessage());
     }
   }
 
+  /**
+   * Update the ex. tax unit price of an invoice line from its in. tax unit price.
+   *
+   * @param request
+   * @param response
+   */
+  public void updatePrice(ActionRequest request, ActionResponse response) {
```

Comment:
Everything in a controller should be in a try/catch block. When catching an exception, use `TracebackService` so the runtime exception can be saved.